plugins/woocommerce/includes/class-wc-order-factory.php

```diff
@@ -28,13 +31,26 @@ public static function get_order( $order_id = false ) {
 			return false;
 		}
 
+		$use_orders_cache = OrderUtil::should_use_orders_cache();
+		if ( $use_orders_cache ) {
+			$order_cache = wc_get_container()->get( OrderCache::class );
+			$order       = $order_cache->get( $order_id );
+			if ( ! is_null( $order ) ) {
+				return $order;
+			}
+		}
+
 		$classname = self::get_class_name_for_order_id( $order_id );
 		if ( ! $classname ) {
 			return false;
 		}
 
 		try {
-			return new $classname( $order_id );
+			$order = new $classname( $order_id );
+			if ( $use_orders_cache ) {
```

Comment:
We should add an `instance of` check for the order class. Some extensions may return a wp_error object if they fail to initialize the object, and we will end up caching the error object in that case.

Reply:
You mean before executing `$order_cache->set`? The `OrderCache::validate` method is already checking if the passed object is an `WC_Abstract_Legacy_Order`. That said, when that's not the case an exception is throw, this will cause `get_order` to return false and that's probably not a good idea.